frontend/app/routes/_public+/apply+/$id+/applicant-information.tsx

```diff
@@ -60,6 +60,14 @@ export async function action({ request, params }: ActionFunctionArgs) {
 }
 
 export default function ApplyFlowApplicationInformation() {
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // redirect to start if the flow has not yet been initialized
+    const flowState = sessionStorage.getItem('flow.state');
+    if (flowState !== 'active') navigate('/apply');
+  }, [navigate]);
+
```

Comment:
Is is possible to have this code in his own component?

Reply:
Yeah. I can work on that. Are you okay with the general approach, though?

- set state in starting route
- check state in non-starting routes
- reset state in end route

Reply:
Idk, if you go with `_layout.tsx` option... check if it render on location change.

Reply:
I think if you go with `_layout.tsx` and `useEffect` it will be good to use `location` as a `useEffect` dependency.
https://remix.run/docs/en/main/hooks/use-location#uselocation

Reply:
The problem with using `_layout.tsx` is that it will fire indiscriminately for all child routes.. but the behavior of the starting route is not the same as the behavior of the non-starting route.

And I don't really want to get into checking the URL to see if it's `=== '/apply'` because that's kinda gross and fragile.

Reply:
`_layout.tsx` will be inside `$id+` directory

Reply:
I added a `routes/_public+/apply+/$id+/_route.tsx` file that contains the `useEffect()`. Please re-review.